.config/gitpod/Dockerfile

```diff
@@ -38,23 +36,37 @@ RUN PYTHON_VERSION=$(cut -d '=' -f 2 /tmp/python_version.config) && \
         vim \
         wget \
     && python3 -m pip install --no-cache-dir --upgrade pip \
-    && python3 -m pip install --no-cache-dir virtualenv \
-    # Install Docker
-    && curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg \
+    && python3 -m pip install --no-cache-dir virtualenv
+
+# Install Docker
+FROM builder as docker-installer
+
+RUN curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg \
```

Comment:
# hadolint ignore=DL3008,DL3009

DL3008 warning: Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`: This is a best practice to ensure build reproducibility to pin the versions of installed packages. However, this can make your Dockerfile more difficult to maintain. If you want to ignore this warning, you can add `# hadolint ignore=DL3008` above the RUN lines.

DL3009 info: Delete the apt-get lists after installing something: You can delete package lists after installation to reduce image size. You are already doing this in the cleanup phase, but hadolint is likely raising this warning for the apt-get update commands in the intermediate stages. You can ignore this warning.

Reply:
@ThomasSanson 

I have just made the modification, and indeed the DL3008 and DL3009 errors for line 44 do not appear during testing.

You can find the modification in this commit: 59c3b6eeb5770fbcc4b5a9d946d47400702f0008